### ISSUE TITLE
Slashes with text ellipsis overflow behaves differently than Android (Desktop)

### ISSUE THREAD

**Issue by JetpackDuba** [JetBrains/compose-multiplatform#1888]
When I have got a `Text` with slashes (like a file path) as text that doesn't fit the current screen, in the Android version of compose the string gets cut partially and not the whole word.  The Desktop version cuts the whole word instead until the slash.

**Desktop:** "FooBar" is not shown, just the ellipsis.

![Desktop](https://user-images.githubusercontent.com/16917468/155236427-339d990b-4c8d-4700-9261-476c7de8cc8c.png)
![ellipsis](https://user-images.githubusercontent.com/16917468/155236997-3131d044-ea41-4715-858a-dcdf49bed6d5.gif)

**Android:** "Foo" is shown + Ellipsis

![Android](https://user-images.githubusercontent.com/16917468/155237453-83117d56-a4e8-42ac-ad23-b8ff2edfa852.png)

Code sample:

```
MaterialTheme {
    Text(
        text = "/Path/To/A/Fantastic/Path/Named/FooBar",
        overflow = TextOverflow.Ellipsis,
        fontSize = 23.sp,
        maxLines = 1,
    )
}
```

`TextOverflow.Clip` behaves in equally in both platform by cutting the whole word.

**Comment by aeab13** [comment]
I'm testing on windows now and surprisingly it works properly there, so seems something related to the linux implementation (no idea about macOS).

**Comment by akurasov** [comment]
Tried on MacOS. It works same way as your desktop example

**Comment by SerVB** [comment]
Also bumped to this. An example from Android: https://developer.android.com/jetpack/compose/text#text-overflow

```kt
@Composable
fun OverflowedText() {
    Text("Hello Compose ".repeat(50), maxLines = 2, overflow = TextOverflow.Ellipsis)
}
```

![grafik](https://user-images.githubusercontent.com/20105593/205075585-af3543e6-b68e-4245-ab3e-e57ede086d8a.png)

But on desktop, it seems it tends to replace the whole word by ellipsis often. I can have `Compo...` but after I reduce the size of the window a little, it becomes not `Comp...` but just `...`. Increasing the size makes it `Compos...` and `Compose...`, so it's not always reproduced.

**Comment by Sage-Kreiter** [comment]
Any updates on this? Seems like very strange behavior to cut off the text at a "/" or "-" on desktop.

Without `overflow = TextOverflow.Ellipsis`:
![image](https://github.com/JetBrains/compose-multiplatform/assets/107443138/cef73d8d-d325-4dd2-bf6f-8ab10157a4bd)

With `overflow = TextOverflow.Ellipsis` and `maxLines = 1`:
![image](https://github.com/JetBrains/compose-multiplatform/assets/107443138/4b13c391-f3ce-4948-9001-6794861bf1c5)

**Comment by dima-avdeev-jb** [comment]
@Sage-Kreiter Can you please provide code snippet to reproduce this problem?

**Comment by Sage-Kreiter** [comment]
Here is a Box with Text that overflows

```
    Box(
        modifier = Modifier
            .size(300.dp)
            .border(width = 1.dp, color = MaterialTheme.colors.secondaryVariant)
    ) {
        Text(
            text = "Baconipsumdolorametjowlfatbackribeyestripsteak,sirloinbiltonghamhock",
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )
    }
```

![image](https://github.com/JetBrains/compose-multiplatform/assets/107443138/e73c70dc-10ee-4e1f-ad0c-efc8b4d8a586)


Here is a Box with text that overflows but contains a space as the 3rd character (same happens with "/" and "-")

```
    Box(
        modifier = Modifier
            .size(300.dp)
            .border(width = 1.dp, color = MaterialTheme.colors.secondaryVariant)
    ) {
        Text(
            text = "Ba onipsumdolorametjowlfatbackribeyestripsteak,sirloinbiltonghamhock",
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )
    }
```

![image](https://github.com/JetBrains/compose-multiplatform/assets/107443138/c3be4de4-a2e7-42fe-876f-9faea8707f21)

**Comment by MatkovIvan** [comment]
This behaviour is defined in skia, Compose itself doesn't control it. It has been already fixed in this commit - https://skia.googlesource.com/skia.git/+/0d31aa1f49cb8cfd790c44d92a9a023839524865
So, it should be fixed once we migrate to new skia version

**Comment by Sage-Kreiter** [comment]
Great. Thanks for looking into this

**Comment by MatkovIvan** [comment]
> Which release can I find the fix?

It will be available in 1.6.0

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-1888">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
